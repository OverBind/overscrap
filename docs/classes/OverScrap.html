<!DOCTYPE html><html class="default"><head><meta charSet="utf-8"/><meta http-equiv="x-ua-compatible" content="IE=edge"/><title>OverScrap | @overbind/overscrap</title><meta name="description" content="Documentation for @overbind/overscrap"/><meta name="viewport" content="width=device-width, initial-scale=1"/><link rel="stylesheet" href="../assets/style.css"/><link rel="stylesheet" href="../assets/highlight.css"/><script async src="../assets/search.js" id="search-script"></script></head><body><script>document.body.classList.add(localStorage.getItem("tsd-theme") || "os")</script><header><div class="tsd-page-toolbar"><div class="container"><div class="table-wrap"><div class="table-cell" id="tsd-search" data-base=".."><div class="field"><label for="tsd-search-field" class="tsd-widget search no-caption">Search</label><input type="text" id="tsd-search-field"/></div><ul class="results"><li class="state loading">Preparing search index...</li><li class="state failure">The search index is not available</li></ul><a href="../index.html" class="title">@overbind/overscrap</a></div><div class="table-cell" id="tsd-widgets"><div id="tsd-filter"><a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a><div class="tsd-filter-group"><div class="tsd-select" id="tsd-filter-visibility"><span class="tsd-select-label">All</span><ul class="tsd-select-list"><li data-value="public">Public</li><li data-value="protected">Public/Protected</li><li data-value="private" class="selected">All</li></ul></div> <input type="checkbox" id="tsd-filter-inherited" checked/><label class="tsd-widget" for="tsd-filter-inherited">Inherited</label><input type="checkbox" id="tsd-filter-externals" checked/><label class="tsd-widget" for="tsd-filter-externals">Externals</label></div></div><a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a></div></div></div></div><div class="tsd-page-title"><div class="container"><ul class="tsd-breadcrumb"><li><a href="../modules.html">@overbind/overscrap</a></li><li><a href="OverScrap.html">OverScrap</a></li></ul><h1>Class OverScrap</h1></div></div></header><div class="container container-main"><div class="row"><div class="col-8 col-content"><section class="tsd-panel tsd-hierarchy"><h3>Hierarchy</h3><ul class="tsd-hierarchy"><li><span class="target">OverScrap</span></li></ul></section><section class="tsd-panel-group tsd-index-group"><h2>Index</h2><section class="tsd-panel tsd-index-panel"><div class="tsd-index-content"><section class="tsd-index-section "><h3>Constructors</h3><ul class="tsd-index-list"><li class="tsd-kind-constructor tsd-parent-kind-class"><a href="OverScrap.html#constructor" class="tsd-kind-icon">constructor</a></li></ul></section><section class="tsd-index-section tsd-is-private tsd-is-private-protected"><h3>Properties</h3><ul class="tsd-index-list"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><a href="OverScrap.html#_page" class="tsd-kind-icon">_page</a></li></ul></section><section class="tsd-index-section "><h3>Methods</h3><ul class="tsd-index-list"><li class="tsd-kind-method tsd-parent-kind-class"><a href="OverScrap.html#getHeroes" class="tsd-kind-icon">get<wbr/>Heroes</a></li><li class="tsd-kind-method tsd-parent-kind-class"><a href="OverScrap.html#getInfo" class="tsd-kind-icon">get<wbr/>Info</a></li><li class="tsd-kind-method tsd-parent-kind-class"><a href="OverScrap.html#getRanks" class="tsd-kind-icon">get<wbr/>Ranks</a></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><a href="OverScrap.html#build" class="tsd-kind-icon">build</a></li></ul></section></div></section></section><section class="tsd-panel-group tsd-member-group "><h2>Constructors</h2><section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class"><a id="constructor" class="tsd-anchor"></a><h3>constructor</h3><ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class"><li class="tsd-signature tsd-kind-icon">new <wbr/>Over<wbr/>Scrap<span class="tsd-signature-symbol">(</span>page<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Page</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="OverScrap.html" class="tsd-signature-type" data-tsd-kind="Class">OverScrap</a></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/OverBind/overscrap/blob/a95a642/src/index.ts#L10">index.ts:10</a></li></ul></aside><h4 class="tsd-parameters-title">Parameters</h4><ul class="tsd-parameters"><li><h5>page: <span class="tsd-signature-type">Page</span></h5></li></ul><h4 class="tsd-returns-title">Returns <a href="OverScrap.html" class="tsd-signature-type" data-tsd-kind="Class">OverScrap</a></h4></li></ul></section></section><section class="tsd-panel-group tsd-member-group tsd-is-private tsd-is-private-protected"><h2>Properties</h2><section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-private"><a id="_page" class="tsd-anchor"></a><h3><span class="tsd-flag ts-flagPrivate">Private</span> _page</h3><div class="tsd-signature tsd-kind-icon">_page<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">Page</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/OverBind/overscrap/blob/a95a642/src/index.ts#L8">index.ts:8</a></li></ul></aside></section></section><section class="tsd-panel-group tsd-member-group "><h2>Methods</h2><section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class"><a id="getHeroes" class="tsd-anchor"></a><h3>get<wbr/>Heroes</h3><ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class"><li class="tsd-signature tsd-kind-icon">get<wbr/>Heroes<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">Hero</span><span class="tsd-signature-symbol">[]</span><span class="tsd-signature-symbol">&gt;</span></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/OverBind/overscrap/blob/a95a642/src/index.ts#L80">index.ts:80</a></li></ul></aside><h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">Hero</span><span class="tsd-signature-symbol">[]</span><span class="tsd-signature-symbol">&gt;</span></h4></li></ul></section><section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class"><a id="getInfo" class="tsd-anchor"></a><h3>get<wbr/>Info</h3><ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class"><li class="tsd-signature tsd-kind-icon">get<wbr/>Info<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">InfoField</span><span class="tsd-signature-symbol">[]</span><span class="tsd-signature-symbol">&gt;</span></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/OverBind/overscrap/blob/a95a642/src/index.ts#L65">index.ts:65</a></li></ul></aside><h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">InfoField</span><span class="tsd-signature-symbol">[]</span><span class="tsd-signature-symbol">&gt;</span></h4></li></ul></section><section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class"><a id="getRanks" class="tsd-anchor"></a><h3>get<wbr/>Ranks</h3><ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class"><li class="tsd-signature tsd-kind-icon">get<wbr/>Ranks<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">Rank</span><span class="tsd-signature-symbol">[]</span><span class="tsd-signature-symbol">&gt;</span></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/OverBind/overscrap/blob/a95a642/src/index.ts#L45">index.ts:45</a></li></ul></aside><h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">Rank</span><span class="tsd-signature-symbol">[]</span><span class="tsd-signature-symbol">&gt;</span></h4></li></ul></section><section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class tsd-is-static"><a id="build" class="tsd-anchor"></a><h3><span class="tsd-flag ts-flagStatic">Static</span> build</h3><ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class tsd-is-static"><li class="tsd-signature tsd-kind-icon">build<span class="tsd-signature-symbol">(</span>username<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span>, hashtag<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span>, platform<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><a href="OverScrap.html" class="tsd-signature-type" data-tsd-kind="Class">OverScrap</a><span class="tsd-signature-symbol">&gt;</span></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/OverBind/overscrap/blob/a95a642/src/index.ts#L14">index.ts:14</a></li></ul></aside><h4 class="tsd-parameters-title">Parameters</h4><ul class="tsd-parameters"><li><h5>username: <span class="tsd-signature-type">string</span></h5></li><li><h5>hashtag: <span class="tsd-signature-type">string</span></h5></li><li><h5>platform: <span class="tsd-signature-type">string</span></h5></li></ul><h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><a href="OverScrap.html" class="tsd-signature-type" data-tsd-kind="Class">OverScrap</a><span class="tsd-signature-symbol">&gt;</span></h4></li></ul></section></section></div><div class="col-4 col-menu menu-sticky-wrap menu-highlight"><nav class="tsd-navigation primary"><ul><li class=""><a href="../modules.html">Exports</a></li></ul></nav><nav class="tsd-navigation secondary menu-sticky"><ul><li class="current tsd-kind-class"><a href="OverScrap.html" class="tsd-kind-icon">Over<wbr/>Scrap</a><ul><li class="tsd-kind-constructor tsd-parent-kind-class"><a href="OverScrap.html#constructor" class="tsd-kind-icon">constructor</a></li><li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><a href="OverScrap.html#_page" class="tsd-kind-icon">_page</a></li><li class="tsd-kind-method tsd-parent-kind-class"><a href="OverScrap.html#getHeroes" class="tsd-kind-icon">get<wbr/>Heroes</a></li><li class="tsd-kind-method tsd-parent-kind-class"><a href="OverScrap.html#getInfo" class="tsd-kind-icon">get<wbr/>Info</a></li><li class="tsd-kind-method tsd-parent-kind-class"><a href="OverScrap.html#getRanks" class="tsd-kind-icon">get<wbr/>Ranks</a></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><a href="OverScrap.html#build" class="tsd-kind-icon">build</a></li></ul></li></ul></nav></div></div></div><footer class="with-border-bottom"><div class="container"><h2>Legend</h2><div class="tsd-legend-group"><ul class="tsd-legend"><li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li><li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private property</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li></ul></div><h2>Settings</h2><p>Theme <select id="theme"><option value="os">OS</option><option value="light">Light</option><option value="dark">Dark</option></select></p></div></footer><div class="container tsd-generator"><p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p></div><div class="overlay"></div><script src="../assets/main.js"></script></body></html>