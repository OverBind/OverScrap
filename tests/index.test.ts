import fs from 'fs';
import path from 'path';

import OverScrap from '../src';

import { GorillaPunch } from './mocks';

describe('Retrieve player data from the overwatch website', () => {
  test('Should get player "GorillaPunch"', async () => {
    const GorillaPunchHtml = fs.readFileSync(
      path.resolve(__dirname, './mocks/html/GorillaPunch.html'),
      'utf8'
    );
    const scrapPlayer = await OverScrap(
      'GorillaPunch',
      '21397',
      'pc',
      GorillaPunchHtml
    );
    console.log(scrapPlayer);
    expect(scrapPlayer).toMatchObject(GorillaPunch);
  });
});
