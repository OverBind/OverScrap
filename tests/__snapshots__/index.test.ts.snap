// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retrieve player data from the overwatch website Should get player "FusGaar" 1`] = `
Object {
  "hashtag": "1912",
  "info": Array [
    Object {
      "detail": "Time Played",
      "value": "10:03:20",
    },
    Object {
      "detail": "Games Lost",
      "value": "25",
    },
    Object {
      "detail": "Games Won",
      "value": "21",
    },
    Object {
      "detail": "Games Played",
      "value": "48",
    },
  ],
  "mapHeroes": Array [
    Object {
      "id": "0x02E00000FFFFFFFF",
      "name": "ALL HEROES",
    },
    Object {
      "id": "0x02E0000000000221",
      "name": "Baptiste",
    },
    Object {
      "id": "0x02E0000000000195",
      "name": "Brigitte",
    },
    Object {
      "id": "0x02E000000000007A",
      "name": "D.Va",
    },
    Object {
      "id": "0x02E0000000000079",
      "name": "Lúcio",
    },
    Object {
      "id": "0x02E000000000013E",
      "name": "Orisa",
    },
    Object {
      "id": "0x02E0000000000007",
      "name": "Reinhardt",
    },
    Object {
      "id": "0x02E0000000000040",
      "name": "Roadhog",
    },
    Object {
      "id": "0x02E000000000023B",
      "name": "Sigma",
    },
    Object {
      "id": "0x02E0000000000009",
      "name": "Winston",
    },
    Object {
      "id": "0x02E00000000001CA",
      "name": "Wrecking Ball",
    },
    Object {
      "id": "0x02E0000000000068",
      "name": "Zarya",
    },
    Object {
      "id": "0x02E0000000000020",
      "name": "Zenyatta",
    },
  ],
  "ranks": Array [
    Object {
      "role": "Tank",
      "sr": "3438",
      "src": "",
    },
    Object {
      "role": "Support",
      "sr": "2626",
      "src": "",
    },
  ],
  "topThree": Array [
    Object {
      "info": Array [
        Object {
          "detail": "Time Played",
          "value": "05:42:26",
        },
        Object {
          "detail": "Games Played",
          "value": "26",
        },
        Object {
          "detail": "Win Percentage",
          "value": "53%",
        },
      ],
      "name": "Reinhardt",
      "src": "",
    },
    Object {
      "info": Array [
        Object {
          "detail": "Time Played",
          "value": "02:12:18",
        },
        Object {
          "detail": "Games Played",
          "value": "11",
        },
        Object {
          "detail": "Win Percentage",
          "value": "52%",
        },
      ],
      "name": "Winston",
      "src": "",
    },
    Object {
      "info": Array [
        Object {
          "detail": "Time Played",
          "value": "31:33",
        },
        Object {
          "detail": "Games Played",
          "value": "3",
        },
        Object {},
      ],
      "name": "Sigma",
      "src": "",
    },
  ],
  "username": "FusGaar",
}
`;

exports[`Retrieve player data from the overwatch website Should get player "Punk" 1`] = `
Object {
  "hashtag": "11600",
  "info": Array [
    Object {
      "detail": "Time Played",
      "value": "35:00:52",
    },
    Object {
      "detail": "Games Lost",
      "value": "76",
    },
    Object {
      "detail": "Games Won",
      "value": "101",
    },
    Object {
      "detail": "Games Played",
      "value": "177",
    },
  ],
  "mapHeroes": Array [
    Object {
      "id": "0x02E00000FFFFFFFF",
      "name": "ALL HEROES",
    },
    Object {
      "id": "0x02E000000000013B",
      "name": "Ana",
    },
    Object {
      "id": "0x02E0000000000200",
      "name": "Ashe",
    },
    Object {
      "id": "0x02E0000000000221",
      "name": "Baptiste",
    },
    Object {
      "id": "0x02E0000000000195",
      "name": "Brigitte",
    },
    Object {
      "id": "0x02E000000000007A",
      "name": "D.Va",
    },
    Object {
      "id": "0x02E0000000000029",
      "name": "Genji",
    },
    Object {
      "id": "0x02E0000000000079",
      "name": "Lúcio",
    },
    Object {
      "id": "0x02E0000000000042",
      "name": "McCree",
    },
    Object {
      "id": "0x02E00000000001A2",
      "name": "Moira",
    },
    Object {
      "id": "0x02E000000000013E",
      "name": "Orisa",
    },
    Object {
      "id": "0x02E0000000000007",
      "name": "Reinhardt",
    },
    Object {
      "id": "0x02E0000000000040",
      "name": "Roadhog",
    },
    Object {
      "id": "0x02E000000000023B",
      "name": "Sigma",
    },
    Object {
      "id": "0x02E0000000000003",
      "name": "Tracer",
    },
    Object {
      "id": "0x02E000000000000A",
      "name": "Widowmaker",
    },
    Object {
      "id": "0x02E0000000000009",
      "name": "Winston",
    },
    Object {
      "id": "0x02E00000000001CA",
      "name": "Wrecking Ball",
    },
    Object {
      "id": "0x02E0000000000068",
      "name": "Zarya",
    },
    Object {
      "id": "0x02E0000000000020",
      "name": "Zenyatta",
    },
  ],
  "ranks": Array [
    Object {
      "role": "Tank",
      "sr": "4685",
      "src": "",
    },
    Object {
      "role": "Damage",
      "sr": "3885",
      "src": "",
    },
    Object {
      "role": "Support",
      "sr": "3900",
      "src": "",
    },
  ],
  "topThree": Array [
    Object {
      "info": Array [
        Object {
          "detail": "Time Played",
          "value": "20:08:31",
        },
        Object {
          "detail": "Games Played",
          "value": "100",
        },
        Object {
          "detail": "Win Percentage",
          "value": "59%",
        },
      ],
      "name": "Sigma",
      "src": "",
    },
    Object {
      "info": Array [
        Object {
          "detail": "Time Played",
          "value": "05:09:01",
        },
        Object {
          "detail": "Games Played",
          "value": "27",
        },
        Object {
          "detail": "Win Percentage",
          "value": "56%",
        },
      ],
      "name": "D.Va",
      "src": "",
    },
    Object {
      "info": Array [
        Object {
          "detail": "Time Played",
          "value": "02:59:28",
        },
        Object {
          "detail": "Games Played",
          "value": "15",
        },
        Object {
          "detail": "Win Percentage",
          "value": "62%",
        },
      ],
      "name": "Zarya",
      "src": "",
    },
  ],
  "username": "Punk",
}
`;
